packages/rum-core/test/common/apm-server.spec.js

```diff
@@ -404,6 +404,40 @@ describe('ApmServer', function () {
     ])
   })
 
+  it('should call beforeSend in _makeHttpRequest', async () => {
+    await apmServer._makeHttpRequest('GET', '/', {
+      payload: 'test',
+      beforeSend: (xhr, url, method, headers, payload) => {
+        expect(typeof xhr.setRequestHeader).toBe('function')
+        xhr.setRequestHeader('custom', 'value')
+        expect(url).toBe('/')
+        expect(method).toBe('GET')
+        expect(payload).toBe('test')
+        expect(headers).toBeUndefined()
+        return true
+      }
+    })
+  })
+
+  it('should reject the request if beforeSend returns falsy', async () => {
+    let promise = apmServer._makeHttpRequest('POST', '/test', {
+      payload: 'test',
+      beforeSend: (xhr, url, method, headers, payload) => {
```

Comment:
let's add a test for async case.

Reply:
Let's decide about the async use-case first.

Reply:
Yeah, we dont need to for now, we can skip and document it.